tutor/src/components/tours/custom/common.jsx

```diff
@@ -20,9 +21,45 @@ function TutorBeta() {
   );
 }
 
+function getClickTarget(clickEvent) {
+  return clickEvent.currentTarget.className.includes('joyride-') && [
+    'A',
+    'BUTTON'
+  ].includes(clickEvent.currentTarget.tagName) ? clickEvent.currentTarget : clickEvent.target;
+}
+
+function bindClickHandler(handlers) {
+
+  return ((clickEvent) => {
+    const el = getClickTarget(clickEvent);
+    const dataType = el.dataset.type;
+
+    let handled = false;
+
+    if (el.className.indexOf('joyride-') === 0) {
+      forEach(handlers, (handler, name) => {
+        if (dataType === name) {
+          handled = true;
+
+          clickEvent.preventDefault();
+          clickEvent.stopPropagation();
+
+          return handler(clickEvent);
+        }
```

Comment:
Are we meaning to end the `forEach` loop early if `handler` returns false?   That's bitten me before, but I'm assuming that's intended since there's the explicit return.